core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -1881,7 +1881,9 @@ public static TableInfo convertGsonToTableInfo(Map<String, String> properties) {
     gsonBuilder.registerTypeAdapter(DataType.class, new DataTypeAdapter());
 
     Gson gson = gsonBuilder.create();
-    TableInfo tableInfo = gson.fromJson(builder.toString(), TableInfo.class);
+    // if the column name contains backslash in the column name, then fromJson will remove that,
+    // so replace like below to keep the "\" in column name and write the proper name in the schema
+    TableInfo tableInfo = gson.fromJson(builder.toString().replace("\\", "\\\\"), TableInfo.class);
```

Comment:
can you add a test case with carbon session

Reply:
actually `QueryTest ` has spark session without `carbonSession`, and we cannot create another `sparkSession `when one is running. In example we cant add as test case, so i think since i have added the proper comment, it should be fine i guess.

Reply:
you can add in CarbonSessionExample

Reply:
ok